### PR TITLE
BFD-2068: Improve SSH user creation

### PR DESCRIPTION
**Pull request by malessi**
**JIRA Ticket:**
[BFD-2068](https://jira.cms.gov/browse/BFD-2068)

**User Story or Bug Summary:**

As a BFD Engineer I want the process for adding new contributors or offboarding contributors that
are leaving to be as simple and robust as possible. Additionally, I want the ability to create users
with the least privilege necessary.

---

### What Does This PR Do?

This PR improves our process of adding SSH users and cleans it up so that all SSH configuration
lives in SSM.

The changes to the build and launch tasks of our instances allow for proper per-environment access
control for each user configured in SSM. A user can be configured to have SSH access globally or
per-environment and likewise with their access to `root` (`sudoer`).

Various unused keys or configuration stores have been removed outright as well, greatly reducing the
risk of confusion regarding where to add or remove SSH configuration.

Additionally, a new runbook, "How to Add or Remove Users' SSH Access", has been added to our
collection of runbooks in the repository. This new runbook details the process for adding and
removing SSH configuration for a given user.

Changes to Terraform have been verified using `terraform plan`. The changes to the `mgmt` module
have been applied as well. The existing set of users with SSH and `sudoers` access have been
converted to this new scheme and been granted the same privileges.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

- Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

- If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

  - Does this PR add any new software dependencies?
    - [ ] Yes
    - [x] No
  - Does this PR modify or invalidate any of our security controls?
    - [ ] Yes
    - [x] No
  - Does this PR store or transmit data that was not stored or transmitted before?
    - [ ] Yes
    - [x] No

- If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
  - Do you think this PR requires additional review of its security implications for other reasons?
    - [ ] Yes
    - [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

- N/A

### Submitter Checklist

I have gone through and verified that...:

- [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
- [x] This PR is reasonably limited in scope, to help ensure that:
  1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
  2. There isn't too much of a burden on reviewers.
  3. Any problems it causes have a small "blast radius".
  4. It'll be easier to rollback if that becomes necessary.
- [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
- [x] The data dictionary has been updated with any field mapping changes, if any were made.
- [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
- [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
- [x] Reviews are requested from both:
  - At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
  - Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
- [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
  - Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by mjburling — DISMISSED** [review]
Well done! This looks to be more complete than the solution I had in mind. 👏

File: runbooks/how-to-add-or-remove-user-SSH-access.md [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+    3. Apply the changes to configuration ensuring that there are no unexpected changes:
+
+       ```bash
+       terraform apply
+       ```
+
+    4. Repeat the above steps for each established environment that was changed
+
+26. Once finished, the new user will be _fully_ provisioned the next time the `Build App AMIs` stage
+    is ran in our deployment pipeline. In the meantime, anytime a _brand new_ instance is _launched_
```
Comment (lsmitchell) [inline runbooks/how-to-add-or-remove-user-SSH-access.md:231]:
```suggestion
    is run in our deployment pipeline. In the meantime, anytime a _brand new_ instance is _launched_
```
Profuse apologizes for my pedantry with past participles (and adjunct alliteration)

**Review by lsmitchell — DISMISSED** [review]
Great work, and good documentation + runbook on how to do everything! Easy to follow even as someone who has little idea how to do any of this.

**Review by lsmitchell — DISMISSED** [review]

File: .secrets [hunk 2]
```diff
@@ -5,12 +5,10 @@ ops/ansible/playbooks-ccs/files/client_data_server_local_test_env_test_keypair.p
 ops/ansible/playbooks-ccs/files/client_data_server_local_test_env_prod_keypair.pem
 ops/ansible/playbooks-ccs/files/client_data_server_local_test_env_dpr_keypair.pem
 ops/ansible/playbooks-ccs/vars/prod/group_vars/all/vault.yml
-ops/ansible/playbooks-ccs/vars/prod/group_vars/all/ssh.yml
 ops/ansible/playbooks-ccs/vars/prod-sbx/group_vars/all/vault.yml
-ops/ansible/playbooks-ccs/vars/prod-sbx/group_vars/all/ssh.yml
 ops/ansible/playbooks-ccs/vars/test/group_vars/all/vault.yml
```
Comment (mjburling) [inline .secrets:9]: Could you also drop the remaining `*/vault.yml` files here? I guess I neglected to do so in the initial implementation.

Comment (malessi) [reply .secrets]: Yep, just did so.

**Review by mjburling — COMMENTED** [review]
Re-iterating my support for this change set, but I think I'd like to see just one more deployment to `test` ahead of accepting this for warmest, fuzziest feelings after your most recent adjustments.

**Comment by malessi** [comment]
> Re-iterating my support for this change set, but I think I'd like to see just one more deployment to `test` ahead of accepting this for warmest, fuzziest feelings after your most recent adjustments.

After some pain with respect to IAM permissions that was causing the `launch*` variant playbooks to fail, the deployment with these changes have succeeded. This PR should _finally_ be set for review.

**Review by mjburling — APPROVED** [review]

**Pull request merged**